### ISSUE TITLE
How to write Backward when  I  write custom layers? Is not Mxnet  Auto-Differentiation? 

### ISSUE THREAD

**Issue by yuantangliang** [dmlc/mxnet#2610]
I am new to mxnet  and I need to custom loss function in Mxnet. I search the google and find that I need to write new operatros,both forward and backward. how to write backward function? what to read and learn?  

by the way,  When I custom loss function in keras ,the only thing I need to do is write the loss function, and pass it to train function. why do I need to write backrward in Mxnet, is not  it  Auto-Differentiation?

any help is appereciate.

**Comment by sxjscience** [comment]
@yuantangliang You need to write backward logic manually for custom layers. Backward pass is automatic only if you are able to build the loss function using basic MXNet symbols (http://mxnet.readthedocs.io/en/latest/packages/python/symbol.html).

**Comment by yuantangliang** [comment]
Thanks you
